### PR TITLE
Add Stepper Input docs

### PR DESCRIPTION
**Pull request by lilyvc**
Adds site docs for Stepper Input.
Will need to be updated after #3976

- [x] theme next QA

### CODE REVIEW

File: site/docs/components/stepper-input/accessibility.mdx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+sidebar:
+  exclude: true
+data:
+  $ref: ./#/data
+---
+
+## Best practices
+
+- Wrap `StepperInput` in [`FormField`](/salt/components/form-field) to provide the field with a visible label, help text and a status message for validation feedback. This will ensure it complies with accessibility guidelines. You can find more information [in the W3C form instructions](https://www.w3.org/WAI/tutorials/forms/instructions/).
+- `aria-valuenow`, `aria-valuemax`, and `aria-valuemin` attributes are set to provide users with information about the current value, and the allowable range of values.
```
Comment (joshwooding) [inline site/docs/components/stepper-input/accessibility.mdx:14]: Is this a best practice? Maybe something to look at in the future. This seems more like a spec of how the component works.

File: site/docs/components/stepper-input/accessibility.mdx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+## Keyboard interactions
+
+Stepper input inherits keyboard interactions from [`Input`](/salt/components/input/accessibility).
+
+The following interactions are specific to Stepper Input.
+
+<KeyboardControls>
+<KeyboardControl keyOrCombos="Up Arrow">
```
Comment (joshwooding) [inline site/docs/components/stepper-input/accessibility.mdx:23]:
```suggestion
<KeyboardControl keyOrCombos="Up arrow">
```

File: site/docs/components/stepper-input/accessibility.mdx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+When the input area has focus, increases the number by the increment value (default to 1), up to the maximum value if set.
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="Shift + Up Arrow">
```
Comment (joshwooding) [inline site/docs/components/stepper-input/accessibility.mdx:28]:
```suggestion
<KeyboardControl keyOrCombos="Shift + Up arrow">
```

File: site/docs/components/stepper-input/accessibility.mdx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+ 
+When the input area has focus, increases the number by the step block value (default to 10), up to the maximum value if set.
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="Page Up">
```
Comment (joshwooding) [inline site/docs/components/stepper-input/accessibility.mdx:33]:
```suggestion
<KeyboardControl keyOrCombos="Page up">
```

File: site/docs/components/stepper-input/accessibility.mdx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+When the input area has focus, increases the number by the step block value (default to 10), up to the maximum value if set.
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="End">
+
+When the input area has focus, increases the number to the maximum value.
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="Down Arrow">
```
Comment (joshwooding) [inline site/docs/components/stepper-input/accessibility.mdx:43]:
```suggestion
<KeyboardControl keyOrCombos="Down arrow">
```

File: site/docs/components/stepper-input/accessibility.mdx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+When the input area has focus, decresase the number by the increment value (default to 1), up to the minimum value if set.
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="Shift + Down Arrow">
```
Comment (joshwooding) [inline site/docs/components/stepper-input/accessibility.mdx:48]:
```suggestion
<KeyboardControl keyOrCombos="Shift + Down arrow">
```

File: site/docs/components/stepper-input/accessibility.mdx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+ 
+When the input area has focus, decreases the number by the step block value (default to 10), up to the maximum value if set.
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="Page Down">
```
Comment (joshwooding) [inline site/docs/components/stepper-input/accessibility.mdx:53]:
```suggestion
<KeyboardControl keyOrCombos="Page down">
```

File: site/docs/components/stepper-input/accessibility.mdx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+## Keyboard interactions
+
+Stepper input inherits keyboard interactions from [`Input`](/salt/components/input/accessibility).
+
+The following interactions are specific to Stepper Input.
+
+<KeyboardControls>
+<KeyboardControl keyOrCombos="Up Arrow">
+
+When the input area has focus, increases the number by the increment value (default to 1), up to the maximum value if set.
```
Comment (joshwooding) [inline site/docs/components/stepper-input/accessibility.mdx:25]: Can't remember which is the correct way but this should say "this action increases" or "Up arrow increases"

File: site/docs/components/stepper-input/accessibility.mdx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="Shift + Up Arrow">
+ 
+When the input area has focus, increases the number by the step block value (default to 10), up to the maximum value if set.
```
Comment (joshwooding) [inline site/docs/components/stepper-input/accessibility.mdx:30]:
```suggestion
When the input area has focus, increases the number by the step block value (defaults to 10), up to the maximum value if set.
```

File: site/src/examples/stepper-input/TextAlignment.tsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import { FormField, FormFieldLabel, StackLayout } from "@salt-ds/core";
+import { StepperInput } from "@salt-ds/lab";
+
+export const TextAlignments = () => (
```
Comment (joshwooding) [inline site/src/examples/stepper-input/TextAlignment.tsx:4]:
```suggestion
export const TextAlignment = () => (
```
This name means the example doesn't show

File: site/src/examples/stepper-input/ButtonAdornments.tsx [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+      const randomDelta = Math.floor(Math.random() * (1000 - 100) + 100) / 100;
+      setRandomLiveValue((prev) => prev + randomDelta);
+    }, 500);
+
+    return () => clearInterval(intervalId);
+  }, []);
+
+  return (
+    <FormField>
+      <FormFieldLabel>Refresh adornment</FormFieldLabel>
```
Comment (joshwooding) [inline site/src/examples/stepper-input/ButtonAdornments.tsx:48]:
```suggestion
      <FormFieldLabel>Sync adornment</FormFieldLabel>
```

File: site/src/examples/stepper-input/Limits.tsx [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  const min = 0;
+
+  const numericValue =
+    typeof value === "number" ? value : Number.parseFloat(value);
+  const isError =
+    typeof value !== "number" || numericValue > max || numericValue < min;
+
+  return (
+    <FormField
+      validationStatus={isError ? "error" : undefined}
```
Comment (joshwooding) [inline site/src/examples/stepper-input/Limits.tsx:17]: I don't think this validation is working correctly, it triggers when I use the increment/decrement buttons

File: site/docs/components/stepper-input/examples.mdx [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+## Default
+
+Stepper input has a default value of 0, an increment value of 1 and a block value of 10, and you can only use it to accept numeric characters. A single click increases or decreases the value in the input by 1.
+
+</LivePreview>
+<LivePreview componentName="stepper-input" exampleName="Variants" >
+
+## Variants
+
+The component comes in two variants: primary and secondary. By default, the primary variant is selected. To switch to the secondary variant, use `variant="secondary"`. When choosing between the primary and secondary variants, consider the background color on which the component will be placed.
```
Comment (joshwooding) [inline site/docs/components/stepper-input/examples.mdx:23]: Not sure about the terminology "the primary variant is selected"

File: site/docs/components/stepper-input/examples.mdx [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+</LivePreview>
+<LivePreview componentName="stepper-input" exampleName="Readonly" >
+
+## Read-only
+
+You can set stepper input to a read-only state. The user can't edit the value text in this state, but they can highlight and copy it.
+
+By default, the marker is an em dash. If your application requires a different marker, you can set it via the `emptyReadOnlyMarker` prop.
+
+- Use the read-only state when the user needs the value for their flow or current task but cannot edit it, for example, because of user permissions.
```
Comment (joshwooding) [inline site/docs/components/stepper-input/examples.mdx:118]: Is this a best practice? There are some short lists without headings (this example and the next one)

File: site/docs/components/stepper-input/examples.mdx [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+data:
+  $ref: ./#/data
+---
+
+<LivePreviewControls>
+<LivePreview componentName="stepper-input" exampleName="Default" >
+
+## Default
+
+Stepper input has a default value of 0, an increment value of 1 and a block value of 10, and you can only use it to accept numeric characters. A single click increases or decreases the value in the input by 1.
```
Comment (joshwooding) [inline site/docs/components/stepper-input/examples.mdx:16]: Can't really test atm but does it actually have a default of 0?

File: site/docs/components/stepper-input/index.mdx [hunk 16]
```diff
@@ -0,0 +1,12 @@
+---
+title: Stepper input
+data:
+  description: "Stepper input displays a default numeric value that users can increase or decrease using the controls or by keyboard actions. Users can also manually enter a specific value. This is particularly useful in scenarios where users need to make small adjustments to a value within a defined range. "
```
Comment (joshwooding) [inline site/docs/components/stepper-input/index.mdx:4]:
```suggestion
  description: "`StepperInput` displays a default numeric value that users can increase or decrease using the controls or by keyboard actions. Users can also manually enter a specific value. This is particularly useful in scenarios where users need to make small adjustments to a value within a defined range. "
```

File: site/docs/components/stepper-input/usage.mdx [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+- When the input is purely numerical and does not require complex formatting.
+- If a user benefits from adjusting values without having to type them manually, especially on touch interfaces where typing can be cumbersome.
+- If there is a requirement to provide immediate visual feedback as the user adjusts the value, helping them understand the impact of their changes in real-time.
+
+### When not to use
+
+- When users need the flexibility to enter any (subjective) value, including those outside a predefined range. Instead, use Input or Multiline input.
+- When users need to select a single date or select a date range, between a start and end date. Instead, use Date input.
+- When users need to select an item from a large list of options without scrolling. Instead, use Combo box.
+- To choose one value from a set of five to ten options that does not require filtering. Instead, use Dropdown.
```
Comment (joshwooding) [inline site/docs/components/stepper-input/usage.mdx:28]: These should probably have links to the components

File: site/docs/components/stepper-input/accessibility.mdx (hunk 8 above)
Comment (navkaur76) [reply site/docs/components/stepper-input/accessibility.mdx]: Let's go with "Up arrow increases"

File: site/docs/components/stepper-input/examples.mdx (hunk 13 above)
Comment (navkaur76) [reply site/docs/components/stepper-input/examples.mdx]: How about "By default, the primary variant is displayed."

File: site/docs/components/stepper-input/usage.mdx [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+
+### Controlled/uncontrolled
+
+Just like native input elements in React, stepper inputs can be controlled or uncontrolled. By default a stepper input is uncontrolled. Use the `defaultValue` prop to control the default value of the stepper input.
+
+If you want to control the value of the stepper input, use the `value` prop. The `onChange` prop is required when using the controlled approach.
+
+## Props
+
+<PropsTable componentName="StepperInput" />
```
Comment (joshwooding) [inline site/docs/components/stepper-input/usage.mdx:44]:
```suggestion
<PropsTable packageName="lab" componentName="StepperInput" />
```

File: site/docs/components/stepper-input/examples.mdx (hunk 15 above)
Comment (origami-z) [reply site/docs/components/stepper-input/examples.mdx]: No, removed line

File: site/src/examples/stepper-input/Limits.tsx (hunk 12 above)
Comment (origami-z) [reply site/src/examples/stepper-input/Limits.tsx:19]: Fixed

File: site/docs/components/stepper-input/accessibility.mdx (hunk 1 above)
Comment (origami-z) [reply site/docs/components/stepper-input/accessibility.mdx]: Removed

File: site/docs/components/stepper-input/examples.mdx [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+## Default
+
+Stepper input has an increment step value of 1 and a step block value of 10, and you can only use it to accept numeric characters.
+
+</LivePreview>
+<LivePreview componentName="stepper-input" exampleName="Variants" >
+
+## Variants
+
+The component comes in two variants: primary and secondary, with default set to primary. To switch to the secondary variant, use `variant="secondary"`. When choosing between the primary and secondary variants, consider the background color on which the component will be placed.
```
Comment (joshwooding) [inline site/docs/components/stepper-input/examples.mdx:23]:
```suggestion
The component comes in two variants: primary and secondary. The primary variant is the default. To switch to the secondary variant, use `variant="secondary"`. 

When choosing between the primary and secondary variants, consider the background color on which the component will be placed.
```

**Review by joshwooding — DISMISSED** [review]

File: site/src/examples/stepper-input/ButtonAdornments.tsx [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+  );
+};
+
+const SyncAdornment = () => {
+  const [randomLiveValue, setRandomLiveValue] = useState(14.75);
+  const [value, setValue] = useState<number | string>(randomLiveValue);
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      const randomDelta = Math.floor(Math.random() * (1000 - 100) + 100) / 100;
+      setRandomLiveValue((prev) => prev + randomDelta);
```
Comment (joshwooding) [inline site/src/examples/stepper-input/ButtonAdornments.tsx:43]: More of a note, when you sync the value you get "invalid" numbers


https://github.com/user-attachments/assets/ace30230-c566-4194-a739-4e4a50f89fa4

**Review by joshwooding — APPROVED** [review]

**Review by mark-tate — APPROVED** [review]

**Pull request merged**